### DISCUSSION TITLE
SelectTable 在 Table 自定义编辑模板选中不刷新值

### DISCUSSION THREAD

**Discussion by wenlino** [dotnetcore/BootstrapBlazor#3820] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

![PixPin_2024-07-07_21-02-46](https://github.com/dotnetcore/BootstrapBlazor/assets/58806067/31d2fa05-dd16-4c92-b970-6de7f0d94c3a)

直接执行无效。尝试调用 StateHasChanged方法后有效。
`
            var w = wieidbRef.GetType().GetMethod("StateHasChanged", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
            w.Invoke(wieidbRef, null);
`

`
        <TableColumn @bind-Field=context.ParentId Visible="false" IsVisibleWhenAdd="true" IsVisibleWhenEdit="true">
            <EditTemplate Context="v">
                <div class="col-12 col-sm-6">
                    <SelectTable TItem="WieIdbOrderList"
                                 OnQueryAsync="t=>GetParentIdSelectQueryData(t,v)"
                                 GetTextCallback="t=>t.OrderName"
                                 OnValueChanged="@(async t=>v.ParentId=t.Id)"
                                 @bind-Value=pidSelectOrderlist
                    @ref=wieidbRef>
                        <TableColumns Context="ss">
                            <TableColumn @bind-Field="ss.Id"></TableColumn>
                            <TableColumn @bind-Field="ss.OrderName"></TableColumn>
                        </TableColumns>
                    </SelectTable>
                </div>
            </EditTemplate>
        </TableColumn>
`

**Comment by bb-auto** [comment]
@wenlino Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@wenlino 看了一下你这没有格式化的代码，大概率不是组件的 `bug` 是你使用不当，复杂组件在 `EditTemplate` 中使用时需要包裹一层，到一个独立组件中完成业务逻辑。官网中有着重强调与示例

**Comment by wenlino** [comment]
![PixPin_2024-07-07_21-02-46](https://github.com/dotnetcore/BootstrapBlazor/assets/58806067/ec98d931-d057-4bcf-b1f9-6479156c9ed7)

Table -> TableColumn ->EditTemplate-> 使用 SelectTable 当选中后界面不会改动。我调用内部 的 StateHasChanged函数有效。

**Comment by wenlino** [comment]
```

<AdminTable TItem="WieIdbOrderList"
            @ref=wieOTeble
            TreeNodeConverter="TreeNodeConverter"
            OnQueryAsync="WieIdbOrderListOnQueryAsync"
            OnDeleteAsync="WieIdbOrderListOnDeleteAsync"
            ExtendButtonColumnWidth="160"
            ClientTableName="OrderAdminTable"
            IsTree>
    <TableColumns>
        <TableColumn @bind-Field=context.ParentId Visible="false" IsVisibleWhenAdd="true" IsVisibleWhenEdit="true">
            <EditTemplate Context="v">
                <div class="col-12 col-sm-6">
                    <SelectTable TItem="WieIdbOrderList"
                                 OnQueryAsync="t=>GetParentIdSelectQueryData(t,v)"
                                 GetTextCallback="t=>t.OrderName"
                                 OnValueChanged="@(async t=>v.ParentId=t.Id)"
                                 @bind-Value=pidSelectOrderlist 
                                 @ref=wieidbRef>
                        <TableColumns Context="ss">
                            <TableColumn @bind-Field="ss.Id"></TableColumn>
                            <TableColumn @bind-Field="ss.OrderName"></TableColumn>
                        </TableColumns>
                    </SelectTable>
                </div>
            </EditTemplate>
        </TableColumn>
    </TableColumns>
</AdminTable>
//原本这样设计，我测试当选中后，界面并不会发生改动。我猜测是当选中后异步并没有执行 StateHasChanged
//所以我后续使用了 
    private SelectTable<WieIdbOrderList> wieidbRef;
    private WieIdbOrderList? _pidSelectOrderlist;
    private WieIdbOrderList? pidSelectOrderlist
    {
        get => _pidSelectOrderlist;
        set
        {
            _pidSelectOrderlist = value;
            var w = wieidbRef.GetType().GetMethod("StateHasChanged", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance); //这里我尝试调用 StateHasChanged方法发现是有效的。
            w.Invoke(wieidbRef, null);
            this.StateHasChanged();
        }
    }
```

**Reply by ArgoZhang** [reply]
@densen2014 老周总结回答一下，我选你的回答做答案

**Reply by densen2014** [reply]
（占位）
